lib/src/models/retry_event_model.dart

```diff
@@ -0,0 +1,14 @@
+import 'dart:async';
+
+/// Retries the `retryFunction` after a set `timeInterval` until `dispose` is called
+class RetryEventModel {
```

Comment:
I don't really get why we need this. Can't we just use `Timer` directly? I guess you want the following:

1. Maintain a list of delayed retries
2. Check if a retry is already pending
3. Cancel a retry

You can all of this do with a Timer:

```dart
_requestEncryptionKeyPending[callParticipant] = Timer(const Duration(...), retryFunction);

_requestEncryptionKeyPending[callParticipant]?.cancel();
_requestEncryptionKeyPending.remove(callParticipant);
```